### ISSUE TITLE
Incorrect and incomplete typings

### ISSUE THREAD

**Issue by MaratFM** [GetStream/stream-chat-js#75]
One of the examples of incorrect typings in index.d.ts

Right now:
```typescript
export class Channel {
  ...
  addMembers(members: User[]): Promise<APIResponse>;
  addModerators(members: User[]): Promise<APIResponse>;
```

Should be:
```typescript
export class Channel {
  ...
  addMembers(members: string[]): Promise<APIResponse>;
  addModerators(members: string[]): Promise<APIResponse>;
```

There are many other cases when typings are different from runtime types.

And overall types are very limited and incomplete, frequent usage of `[propName: string]: any;` and `any` makes typings useless. Integrating Stream chat into Typescript codebase is very difficult.

**Comment by vishalnarkhede** [comment]
Hey @MaratFM 

Basic issues with our declaration typescript have been fixed (v1.0.1). We are working on adding stronger types on our side. They will be available in next week.

Thanks for bringing this up :)

**Comment by vishalnarkhede** [comment]
Hey @MaratFM 

Stronger types have been added now, please use v1.1.1

Feedback is always appreciated :)
